gsl/span

```diff
@@ -630,7 +630,7 @@ template <class ElementType, std::ptrdiff_t Extent>
 constexpr ElementType& at(const span<ElementType, Extent>& s, size_t index)
 {
     // No bounds checking here because it is done in span::operator[] called below
-    return s[index];
+    return s[narrow_cast<typename span<ElementType, Extent>::index_type>(index)];
```

Comment:
Since the other PR changes the index type to std::ptrdiff_t, there is no need for this cast anymore. Once the other PR is merged, I will clean this one up so that only the fixes that are needed there.